common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -329,6 +329,44 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](dbProtocol: St
         transid.failed(this, start, s"[QUERY] '$dbName' internal error, failure: '${failure.getMessage}'", ErrorLevel))
   }
 
+  protected[core] def count(table: String, startKey: List[Any], endKey: List[Any], skip: Int, stale: StaleParameter)(
+    implicit transid: TransactionId): Future[Long] = {
+
+    val (realStartKey, realEndKey) = (startKey, endKey)
+    val parts = table.split("/")
+
+    val start = transid.started(this, LoggingMarkers.DATABASE_QUERY, s"[COUNT] '$dbName' searching '$table")
+
+    val f = for (eitherResponse <- client.executeView(parts(0), parts(1))(
```

Comment:
Is there a special  reason why you are using a for-comprehension instead of map?

Reply:
Followed the other methods in the file. It’s the same just easier to read in some cases.

Reply:
A map would be easier to follow here though.